appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt

```diff
@@ -1,35 +1,39 @@
 package org.kinecosystem.appsdiscovery.sender.discovery.presenter
 
 import org.kinecosystem.appsdiscovery.base.BasePresenter
-import org.kinecosystem.appsdiscovery.base.OperationResultCallback
 import org.kinecosystem.appsdiscovery.sender.discovery.view.IAppsDiscoveryListView
 import org.kinecosystem.appsdiscovery.sender.model.EcosystemApp
-import org.kinecosystem.appsdiscovery.sender.server.Network
+import org.kinecosystem.appsdiscovery.sender.repositories.DiscoveryAppsRepository
+import java.util.*
 
-class AppsDiscoveryListPresenter : BasePresenter<IAppsDiscoveryListView>(), IAppsDiscoveryListPresenter {
-    override fun loadData() {
-        val network = Network()
-        network.getDiscoveryApps()
-//        network.getDiscoveryApps(object : OperationResultCallback<List<EcosystemApp>?> {
-//            override fun onResult(apps: List<EcosystemApp>?) {
-//                apps?.let {
-//                    view?.updateData(apps)
-//                }
-//            }
-//
-//            override fun onError(errorCode: Int) {
-//            }
-//        })
+class AppsDiscoveryListPresenter(private val discoveryAppsRepository: DiscoveryAppsRepository) : BasePresenter<IAppsDiscoveryListView>(), IAppsDiscoveryListPresenter, Observer {
```

Comment:
I thought we were not using 'I' as naming convention for interfaces

Reply:
as i see in the core it has